packages/ra-core/src/auth/useCanAccess.ts

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useMemo } from 'react';
+import {
+    QueryObserverResult,
+    useQuery,
+    UseQueryOptions,
+} from '@tanstack/react-query';
+import useAuthProvider from './useAuthProvider';
+import useLogoutIfAccessDenied from './useLogoutIfAccessDenied';
+import { useEvent } from '../util';
+
+/**
+ * A hook that calls the authProvider.canAccess() method using react-query for a provided resource and action (and optionally a record).
+ *
+ * The return value updates according to the request state:
+ *
+ * - start: { isPending: true }
+ * - success: { canAccess: true, isPending: false }
+ * - error: { error: [error from provider], isPending: false }
+ *
+ * Useful to enable or disable features based on users permissions.
+ *
+ * @param {Object} params Any params you want to pass to the authProvider
+ * @param {string} params.resource The resource to check access for
+ * @param {string} params.action The action to check access for
+ * @param {Object} params.record Optional. The record to check access for
+ *
+ * @returns Return the react-query result and a canAccess property which is a boolean indicating the access status
+ *
+ * @example
+ *     import { useCanAccess } from 'react-admin';
+ *
+ *     const PostDetail = () => {
+ *         const { isPending, canAccess } = useCanAccess({
+ *             resource: 'posts',
+ *             action: 'read',
+ *         });
+ *         if (isPending || canAccess) {
+ *             return null;
+ *         }
+ *         return <PostEdit />;
+ *     };
+ */
+export const useCanAccess = <ErrorType = Error>(
+    params: UseCanAccessOptions<ErrorType>
+): UseCanAccessResult<ErrorType> => {
+    const authProvider = useAuthProvider();
+    const logoutIfAccessDenied = useLogoutIfAccessDenied();
+    const { onSuccess, onError, onSettled, ...queryOptions } = params ?? {};
+
+    const result = useQuery({
+        queryKey: ['auth', 'canAccess', JSON.stringify(params)],
+        queryFn: async ({ signal }) => {
+            if (!authProvider || !authProvider.canAccess) {
+                return true;
+            }
+            return authProvider.canAccess({
+                ...params,
+                signal,
```

Comment:
we've had to introduce `supportAbortSignal` in #9901 to avoid problems in strict mode. I fear we may have to do the same here.